### ISSUE TITLE
c5_m5_checks_script.sh - modifies fstab when it shouldn't

### ISSUE THREAD

**Issue by dthvt** [awslabs/aws-support-tools#75]
Line 74:

```
        sed -i "s|^/dev/${LINE}|UUID=${UUID}|" /etc/fstab               # Changes the entry in fstab to UUID form
```

Modifies /etc/fstab in place instead of making changes in a temp copy.

Line 80-106

```
        echo -e "\n\nERROR  Your fstab file contains device names. Mount the partitions using UUID's before changing an instance type to M5/C5."                                                         # Outputs the new fstab file
        printf "\nPress y to replace device names with UUID in your fstab file? (y/n) "
        read RESPONSE;
        case "$RESPONSE" in
            [yY]|[yY][eE][sS])                                              # If answer is yes, keep the changes to /etc/fstab
                    echo "Writing changes to /etc/fstab..."
                    echo -e "\n\n***********************"
                    cat /etc/fstab
                    echo -e "***********************"
                    echo -e "\nOriginal fstab file is stored as /etc/fstab.backup.$time_stamp"
                    ;;
            [nN]|[nN][oO]|"")                                               # If answer is no, or if the user just pressed Enter
                    echo -e "Aborting: Not saving changes...\nPrinting correct fstab file below:"                  # don't save the new fstab file
                    cat /etc/fstab
                    cp /etc/fstab.backup.$time_stamp /etc/fstab
                    rm /etc/fstab.backup.$time_stamp
                    ;;
            *)                                                              # If answer is anything else, exit and don't save changes
                    echo "Invalid Response"                                 # to fstab
                    echo "Exiting"
                    cp /etc/fstab.backup.$time_stamp /etc/fstab
                    rm /etc/fstab.backup.$time_stamp
                    exit 1
                    echo -e "------------------------------------------------"
                    ;;
    
        esac
```

Prompts the user to accept changes, however if user kills script while at line 82, /etc/fstab will be modified and not rolled back.  Script output is explicitly wrong and misleading - e.g. line 85 implies the /etc/fstab modification is taking place at that point, when in fact it's already been done w/o user consent.

Script should make changes in a temp file and only modify /etc/fstab when explicitly accepted by user.

**Comment by lpierillas** [comment]
Hi,

Thank you for the feedback. As mentioned having the script make change in a temp file first would make more sense and the output seems misleading. I'll mark this as a bug and request the owner of the tool to investigate this.

**Comment by lpierillas** [comment]
Should be fixed by #92
